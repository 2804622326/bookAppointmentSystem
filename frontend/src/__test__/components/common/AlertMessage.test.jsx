import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import AlertMessage from '../../../components/common/AlertMessage.jsx';

test('renders alert when message provided', () => {
  const { asFragment } = render(<AlertMessage type="success" message="Saved" />);
  expect(screen.getByText('Saved')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('returns null when no message', () => {
  const { container } = render(<AlertMessage type="info" message="" />);
  expect(container.firstChild).toBeNull();
});
