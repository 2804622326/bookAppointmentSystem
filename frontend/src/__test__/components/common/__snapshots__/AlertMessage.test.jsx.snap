// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders alert when message provided 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-success alert-dismissible show"
    role="alert"
  >
    <button
      aria-label="Close alert"
      class="btn-close"
      type="button"
    />
    Saved
  </div>
</DocumentFragment>
`;
